python/cudf/cudf/core/dataframe.py

```diff
@@ -3773,7 +3773,7 @@ def agg(self, aggs, axis=None):
             cols=aggs.keys()
             if any([callable(val) for val in aggs.values()]):
                 raise NotImplementedError("callable parameter is not implemented yet")
-            elif any([isinstance(val, list) for val in aggs.values()]):
+            elif any([isinstance(val, Iterable) and not isinstance(val, str) for val in aggs.values()]):
```

Comment:
Actually this is not what you would expect for this branch, here you want to capture the set of dictionaries whose `values` are either `Iterable` or `str`.

Swapping the order of the branches a bit would achieve this, by first checking:
` elif all([isinstance(val, str) for val in aggs.values()]):` then `elif any([isinstance(val, Iterable) and not isinstance(val, str) for val in aggs.values()]):`

Reply:
...And should satisfy that all `values` are iterables, not just any.
```suggestion
            elif all([isinstance(val, Iterable) for val in aggs.values()]):
```